sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodsCheck.java

```diff
@@ -0,0 +1,281 @@
+        return new int[] {
+            TokenTypes.METHOD_CALL,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.METHOD_CALL:
+                final DetailAST dot = ast.getFirstChild();
+                // method that looks like: method()
+                final String methodNameInCode;
+                if (dot.getType() == TokenTypes.IDENT) {
+                    methodNameInCode = dot.getText();
+                }
+                // method that looks like: obj.method()
+                else {
+                    methodNameInCode = dot.getLastChild().getText();
+                }
+                final int numArgsInCode = dot.getNextSibling().getChildCount(TokenTypes.EXPR);
+                checkForbiddenMethod(ast, methodNameInCode, numArgsInCode);
+                break;
+            default:
+                Utils.reportInvalidToken(ast.getType());
+                break;
+        }
+    }
+
+    /**
+     * Check if the method/constructor call against defined rules.
+     * @param ast ast for the call
+     * @param name ruleName of the the method
+     * @param argCount number of arguments of the method
+     */
+    private void checkForbiddenMethod(DetailAST ast, String name, int argCount) {
+        if (methodName != null) {
+            final boolean methodNameMatches = methodName.matcher(name).matches();
+            boolean argCountMatches = argumentCountRanges.isEmpty();
+            for (IntRange intRange : argumentCountRanges) {
+                if (intRange.contains(argCount)) {
+                    argCountMatches = true;
+                }
+            }
+            if (methodNameMatches && argCountMatches) {
+                if (argumentCountRanges.isEmpty()) {
+                    log(ast.getLineNo(), ast.getColumnNo(), MSG_KEY_WITHOUT_ARG, name, methodName);
+                }
+                else {
+                    log(ast.getLineNo(), ast.getColumnNo(), MSG_KEY_WITH_ARG,
+                        name, methodName,
+                        argCount, argumentCount);
+                }
+            }
+        }
+    }
+
+    /**
+     * Represents a range of non-negative integers.
+     * Range must be bounded on one side or both sides.
+     * It can't be unbounded on both side.
+     * <br>
+     * Some examples of vaild ranges:
+     * <ul>
+     * <li>1-10: 1 and 10 and all numbers between 1 and 10</li>
+     * <li>-10: same as 0-10</li>
+     * <li>5-: same as 5-infinity</li>
+     * <li>1: same as 1-1</li>
+     * </ul>
+     */
+    static class IntRange {
+        /** Regex for matching range. */
+        private static final Pattern RANGE_PATTERN =
+            Pattern.compile("^\\s*+(\\d*+)\\s*+-\\s*+(\\d*+)\\s*+$");
```

Comment:
What is `\\s*+` and `(\\d*+)` exactly saying?
Should this be `\\s+?` and `(\\d+)?` respectively?

Reply:
`\\s*+` & `\\d*+` are Possessive quantifiers (will not match a number partially) and are applicable in this scenario. If we use lazy quantifiers `\\d+?` then in a 2 digit number it will match only the first digit (in 42 will match only 4). Possessive quantifier are more readable and more efficient.
![image](https://user-images.githubusercontent.com/1521534/33696209-fb2000d8-dab5-11e7-826c-f0cd980e3e85.png)


For `-` outside of character class (`[]`),  `-` is matched literally.

Reply:
Thanks for the info. I will look into this more.